File: CPE/Exam/241015/pD_UVA10539.cpp
#include <bits/stdc++.h>
using namespace std;
using LL = long long;
const LL MAXN = 1e12;
const LL MAXN_SQRT = sqrt(MAXN);
#define endl '\n'
#define all(x) x.begin(), x.end()

vector<LL> almost_primes;
vector<bool> is_prime(MAXN_SQRT + 1, true);

void init() {
    is_prime[0] = is_prime[1] = false;
    for (LL i = 2; i * i <= MAXN_SQRT; i++) {
        if (is_prime[i]) {
            for (LL j = i * i; j <= MAXN_SQRT; j += i) {
                is_prime[j] = false;
            }
        }
    }
    for (LL i = 2; i <= MAXN_SQRT; i++) {
        if (is_prime[i]) {
            LL x = i * i;
            while (x <= MAXN) {
                almost_primes.push_back(x);
                x *= i;
            }
        }
    }
    sort(all(almost_primes));
}

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    init();
    int t;
    cin >> t;
    while (t--) {
        LL lo, hi;
        cin >> lo >> hi;
        cout << upper_bound(all(almost_primes), hi) - lower_bound(all(almost_primes), lo) << endl;
    }
    return 0;
}

File: CPE/Exam/241015/pD_UVA10539.py
from math import isqrt
from bisect import bisect_left, bisect_right

MAXN = int(1e12)
MAXN_SQRT = isqrt(MAXN)
is_prime = [True] * (MAXN_SQRT + 1)
is_prime[0] = is_prime[1] = False
for i in range(2, MAXN_SQRT + 1):
    if is_prime[i]:
        for j in range(i * i, MAXN_SQRT + 1, i):
            is_prime[j] = False

almost_primes = []
for i in range(2, MAXN_SQRT + 1):
    if is_prime[i]:
        x = i * i
        while x <= MAXN:
            almost_primes.append(x)
            x *= i
almost_primes.sort()

t = int(input())
for _ in range(t):
    lo, hi = map(int, input().split())
    print(bisect_right(almost_primes, hi) - bisect_left(almost_primes, lo))